compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/CallExpressionResolver.kt

```diff
@@ -213,6 +219,12 @@ class CallExpressionResolver(
             CheckArgumentTypesMode.CHECK_VALUE_ARGUMENTS,
             initialDataFlowInfoForArguments
         )
+
+        // Record call and resolved call in the trace because it is recorded by the
+        // temporary trace above only.
+        context.trace.record(BindingContext.CALL, callExpression, call)
+        context.trace.record(BindingContext.RESOLVED_CALL, call, resolvedCall)
+
```

Comment:
Alternatively we could call the typeResolutionInterceptor with the temporary context or explicitly hand it the call and resolvedCall.